rfcs/0029-allow-script-multiple-matches-on-identical-code/0029-allow-script-multiple-matches-on-identical-code.md

```diff
@@ -1 +1,93 @@
-In review, see <https://github.com/nervosnetwork/rfcs/pull/222>
+---
+Number: "0029"
+Category: Consensus (Hard Fork)
+Status: Draft
+Author: Ian Yang
+Organization: Nervos Foundation
+Created: 2021-02-03
+---
+
+# Allow Multiple Cell Dep Matches When There Is No Ambiguity
+
+## Abstract
+
+This document proposes a transaction verification consensus change to allow multiple cell dep matches on type script hash when all the matches are resolved to the same script code.
+
+## Motivation
+
+Currently, CKB does not allow multiple matches when resolving the script by type script hash in dep cells.
+
+However, in many cases, multiple matches do not introduce ambiguity.
+
+There are two different ways to reference the script, via data hash or type hash.
```

Comment:
`reference` here is a little vague, it could mean 2 different situations:

* When CKB locates the code for lock/type script to execute
* When an on-chain script tries to find a cell containing certain data

I think this RFC only deals with the first kind, it's better if we be more precise and clarify this.